### DISCUSSION TITLE
HA-Addon: Geräte nach HA-Neustart nicht erreichbar

### DISCUSSION THREAD

**Discussion by d33c0n** [OpenCCU/OpenCCU#2761] [🇩🇪 Fragen&Antworten] [answered]
Hallo zusammen,

ich habe Raspberrymatic als HomeAssistant addon (3.75.7.20240601), das wiederum in Form von HAOS 12.3 in einer VM unter Proxmox auf einem x86 Server läuft. Funkmodul ist der HmIP-RFUSB mit der FW 4.4.18. Prinzipiell läuft dieses setup stabil.

Jetzt habe ich das Problem, dass nach einem Neustart der VM der angschlossene HMIP-RFUSB zwar erkannt wird, dieser aber keine Verbindung mehr zu den verbundenen Geräten hat, d.h. Schaltbefehle etc. laufen ins leere, es werden keine Statusmeldungen mehr empfangen usw. Das betrifft auch das Webinterface von RaspberryMatic, nicht nur die HA-Oberfläche. Lokale Bedienung am Gerät funktionieren, führen aber nicht dazu dass die Verbindung wieder hergestellt wird.

Ein weiterer Neustart des RaspberryMatic-Addons behebt das Problem, und die Geräte werden wieder angesprochen.

In den Log vom Addon sieht alles gut aus, der HMIP_RFUSB wird erkannt, auch sonst finde ich keine Fehlermeldungen die mir weiterhelfen (bzw. einfach überhaupt keine Fehlermeldungen), keine Alarm- oder Servicemeldungen. Die Verbindung zwischen HA und der virtuellen CCU scheint auch fehlerfrei zu sein.

Ich beobachte das jetzt seit einer weile, anfangs dachte ich da hätte sich nur zufällig etwas verklemmt, aber das beschriebene Verhalten tritt jetzt seit einer Weile konsistent bei jedem VM-Neustart auf, mittlerweile bin ich Ratlos... Ich hab leider auch per Suche keine anderen Hinweise auf das Problem finden können.

Hat jemand eine Idee wie ich weiter debuggen könnte bzw. was der Fehler sein könnte?

Edit: typos

**Comment by d33c0n** [comment]
Wirklich niemand hier hat eine ähnliche Symptomatik beobachtet?

**Reply by frver** [reply]
Just to let you know, I experience the same problem. For example when updating the haos kernel, raspberrymatic can not find any device. To recover I restart the raspberrymatic plugin, and after that I need to restart the Homematic IP plugin. After that it is working again. Happens quite often. (Also in my case haos is running under proxmox)
Will look into the suggestion by jens-maus. But I did move to haos in the hope to get less issues (I had disconnection issues between HA and raspbery matic, when both HA and raspberry matic where running in docker containers when restarting one or the other (after updates))

**Comment by jens-maus** [comment, marked as answer]
> ich habe Raspberrymatic als HomeAssistant addon (3.75.7.20240601), das wiederum in Form von HAOS 12.3 in einer VM unter Proxmox auf einem x86 Server läuft. Funkmodul ist der HmIP-RFUSB mit der FW 4.4.18. Prinzipiell läuft dieses setup stabil.

"Best Practice" wäre – wenn du schon Proxmox ohnehin einsetzt – RaspberryMatic nicht innerhalb von HA als Addon laufen zu lassen, sondern als separate OVA-basierte VM. Du hast absolut keinerlei vorteil davon RaspberryMatic als Addon in HA laufen zu lassen wenn du schon Proxmox einsetzt. Eher umgedreht, damit handelst du dir unnötigerweise eine zusätzliche Virtualisierungseben ein die nicht sein muss. Ich würde dir also raten statt des Addons eine separate RaspberryMatic VM parallel zu HAos laufen zu lassen und diese dann einfach in der Homematic(IP) Local integration anzugeben. Damit erleichterst du dir das Leben allgemein!
 
> Jetzt habe ich das Problem, dass nach einem Neustart der VM der angschlossene HMIP-RFUSB zwar erkannt wird, dieser aber keine Verbindung mehr zu den verbundenen Geräten hat, d.h. Schaltbefehle etc. laufen ins leere, es werden keine Statusmeldungen mehr empfangen usw. Das betrifft auch das Webinterface von RaspberryMatic, nicht nur die HA-Oberfläche. Lokale Bedienung am Gerät funktionieren, führen aber nicht dazu dass die Verbindung wieder hergestellt wird.
> 
> Ein weiterer Neustart des RaspberryMatic-Addons behebt das Problem, und die Geräte werden wieder angesprochen.

Das sind genau diese von mir angesprochenen "Probleme" mit einer zusätzlichen Virtualsierungseben. Da kann es hier+da eben auch mal haken. Nimm Abstand vom Addon und installier RaspberryMatic als vollwertige VM. Dann kannst du beide Systeme auch völlig unabhängig voneinander warten, backupen, betreiben...

> In den Log vom Addon sieht alles gut aus, der HMIP_RFUSB wird erkannt, auch sonst finde ich keine Fehlermeldungen die mir weiterhelfen (bzw. einfach überhaupt keine Fehlermeldungen), keine Alarm- oder Servicemeldungen. Die Verbindung zwischen HA und der virtuellen CCU scheint auch fehlerfrei zu sein.

Wenn du wirklich weiterhin das Addon innerhalb von HA einsetzen willst, dann bräuchten wir hier mehr Infos. Was sagt der DC bzw. CS wenn das passiert? Was sagen die logfiles? usw.

**Comment by Baxxy13** [comment]
Ich kann hier Jens nur zustimmen, "doppelte Virtualisierung" sollte man möglichst vermeiden.
Auch ich würde mich vom RM-HA-AddOn verabschieden und RM als eigenständige VM auf dem Proxmox laufen lassen.

Da hier auch keinerlei Logfiles oder Screenshots zu sehen sind kann man nicht mal mutmaßen was das Problem sein könnte...

**Comment by d33c0n** [comment]
Hi zusammen, vielen Dank für eure Antworten. Bin überzeugt, werde RM dann separat laufen lassen. Ich hatte gedacht dass das so einfacher zu administrieren ist, aber das Argument mit der doppelten Virtualisierung macht Sinn.

Nur der Vollständigkeit halber: DC ist bei 1-2%, CS bei 0%. Im Log habe ich nichts aufschlussreiches gefunden (s.u.). Gibt es noch andere log files?

![grafik](https://github.com/jens-maus/RaspberryMatic/assets/120460933/25f4a9cd-8286-4f66-8712-dacdab25fce3)
![grafik](https://github.com/jens-maus/RaspberryMatic/assets/120460933/7b9febf2-0bfb-45e8-9736-96f2040f5791)

**Comment by Munic-KW** [comment]
Ich habe das gleiche problem obwohl raspberrsmatic in Promox vm läuft

**Comment by akcasoy** [comment]
Ich habe auch das gleiche problem obwohl Raspberrymatic in Proxmox vm läuft (Firmwareversion: 3.77.7.20240826).

Gestern Nacht wegen Proxmox Backup (stop mode) wurde der VM automatisch runter und wieder hochgefahren. Nach ein paar Stunden haben die Geräte sich nachhinein immer wieder "Kommunikationsfehler" gemeldet. Nach einem manuellen Neustart hatte ich wieder gar kein Fehler.

Das Problem habe ich sehr selten, aber passiert so jeden dritten/vierten Monat gefühlt..
